### PR TITLE
[Fix](Json) fix some cast issue

### PR DESCRIPTION
**Pull request by eldenmoon**
1. casting to boolean/double type should consider all numeric types including int , double and boolean
2. casting to floating point type should consisder boolean types and int types

## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

File: be/src/vec/functions/function_cast.h [hunk 1]
```diff
@@ -856,12 +856,19 @@ struct ConvertImplFromJsonb {
...
                 }
-
                 if constexpr (type_index == TypeIndex::UInt8) {
+                    // cast from json value to boolean type
                     if (value->isTrue()) {
                         res[i] = 1;
                     } else if (value->isFalse()) {
                         res[i] = 0;
+                    } else if (value->isInt()) {
+                        res[i] = ((const JsonbIntVal*)value)->val() > 0 ? 1 : 0;
```
Comment (xiaokang) [inline be/src/vec/functions/function_cast.h:866]: It more common to use `val() == 0 ? 0 : 1`. And `CAST(-1 AS BOOLEAN) ` return 1 in Doris.

File: be/src/vec/functions/function_cast.h [hunk 2]
```diff
@@ -856,12 +856,19 @@ struct ConvertImplFromJsonb {
...
+                    } else if (value->isDouble()) {
+                        res[i] = static_cast<ColumnType::value_type>(
+                                         ((const JsonbDoubleVal*)value)->val()) > 0
```
Comment (xiaokang) [inline be/src/vec/functions/function_cast.h:869]: == 0

File: be/src/vec/functions/function_cast.h (hunk 1 above)
Comment (eldenmoon) [reply be/src/vec/functions/function_cast.h]: done

File: be/src/vec/functions/function_cast.h (hunk 2 above)
Comment (eldenmoon) [reply be/src/vec/functions/function_cast.h]: done

**Comment by eldenmoon** [comment]
run buildall

**Review by amorynan — APPROVED** [review]
LGTM

**Comment by eldenmoon** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Comment by eldenmoon** [comment]
not merge 2.0, reason 1. behavior change 2. no user encountered only in variant cases

**Pull request merged**